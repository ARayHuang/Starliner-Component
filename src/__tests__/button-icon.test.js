import React from 'react';
import { shallow, mount } from 'enzyme';
import IconButton from '../components/button-icon';

describe('IconButton', () => {
	it('should render correctly', () => {
		const props = {
			className: 'mock-button-className',	
		};

		const wrapper = shallow(
			<IconButton {...props}>
				<div>mock-children</div>
			</IconButton>
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-button-className',
			variant: 'outlined',
			size: 'sm',
			isDisabled: false,
			text: 'mock-text',
			icon: <mock-icon />,
			onClick: () => {}		
		};
		
		const wrapper = mount(
			<IconButton {...props}>
				mock-text
			</IconButton>
		);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().variant).toEqual(props.variant);
		expect(wrapper.props().size).toEqual(props.size);
		expect(wrapper.props().isDisabled).toEqual(props.isDisabled);
		expect(wrapper.props().icon).toEqual(props.icon);
		expect(wrapper.props().children).toEqual(props.text);
	})
})
