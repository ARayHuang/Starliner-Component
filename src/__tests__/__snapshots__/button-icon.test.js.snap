// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render correctly 1`] = `
<Fragment>
  <IconButton
    border="1px solid transparent"
    boxShadow="0px 3px 1px -2px rgba(#000, 0.2), 0px 2px 2px 0px rgba(#000, 0.14), 0px 1px 5px 0px rgba(#000, 0.12)"
    className="mock-button-className"
    isRound={true}
    size="sm"
  />
</Fragment>
`;
